compiler/src/dmd/dsymbolsem.d

```diff
@@ -1037,9 +1037,24 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
                             ex = (cast(AssignExp)ex).e2;
                         if (auto ne = ex.isNewExp())
                         {
-                            // See if initializer is a NewExp that can be allocated on the stack
+                            /* See if initializer is a NewExp that can be allocated on the stack.
+                             */
                             if (dsym.type.toBasetype().ty == Tclass)
                             {
+                                /* Unsafe to allocate on stack if constructor is not `scope` because the `this` can leak.
+                                 * https://issues.dlang.org/show_bug.cgi?id=23145
+                                 */
+                                if (ne.member && !(ne.member.storage_class & STC.scope_))
+                                {
+                                    if (sc.func.isSafe())
+                                    {
+                                        deprecation(dsym.loc,
```

Comment:
Add common deprecation comment to later deprecate after the deprecation period

Reply:
I don't know how long the period should be. Depends on how disruptive this is.

Reply:
I think the default is 10 releases, but I indeed believe it's too long. We need to state this in the code before merging to avoid forgetting.

Reply:
grepping for  "deprecation" should do the trick.

Reply:
@WalterBright grep for `@@@` and you'll see the comment style used to flag them.

Standard deprecation length is 10 releases, for extreme cases this is 20 releases.

Reply:
added